File: Java/55. Jump Game.java
class Solution {
    public boolean canJump(int[] nums) {
        int max = 0;
        while (max < nums.length - 1) {
            int tmp = max;
            for (int i = 0; i < tmp + 1; i++) {
                max = Math.max(nums[i] + i, max);
            }
            if (tmp == max && max < nums.length)
                return false;
        }
        return true;
    }
}

File: 2023/55. Jump Game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int last = 0;
        for (int i = 0; i < nums.size() && i <= last && last + 1 < nums.size(); i++) {
            last = max(i + nums[i], last);
        }
        return last + 1 >= nums.size();
    }
};